shell/platform/darwin/ios/framework/Source/FlutterPluginAppLifeCycleDelegate.mm

```diff
@@ -32,6 +38,33 @@ @implementation FlutterPluginAppLifeCycleDelegate {
   NSPointerArray* _delegates;
 }
 
+static void RemapMethod(Class thisClass, SEL originalSelector, SEL addedSelector) {
+  Method originalMethod = class_getInstanceMethod(thisClass, originalSelector);
+  IMP originalImp = method_getImplementation(originalMethod);
+  class_addMethod(thisClass, addedSelector, originalImp, method_getTypeEncoding(originalMethod));
+}
+
++ (void)load {
+  // Swap out
+  // `performApplication:didReceiveRemoteNotification:fetchCompletionHandler:`
+  // for `application:didReceiveRemoteNotification:fetchCompletionHandler:`.
+  // This has to be done to avoid a potential false positive "Missing Push
+  // Notification Entitlement" validation failure when uploading apps to the App
+  // Store.  Linked NSPlugin implementations that have
+  // `application:didReceiveRemoteNotification:fetchCompletionHandler:` defined
+  // will cause the failure to happen correctly if the entitlements are missing.
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    RemapMethod(
+        [self class],
```

Comment:
Very small nit `self` is already the `class` in a class method.
```suggestion
        self,
```

Reply:
done